[Banner Ads] Podcast list

bjtitus opened the pull request:
| 📘 Part of: #3208 |
|:---:|

Fixes #3210

* Adds the `BannerAdView` SwiftUI view
* Color themes for the podcast list - a different one will be used for the player
* Disables the `banner_ads` feature flag for now so we don't ship anything with the static banner ads.
* Ad fetching is still to come so the ad strings don't need to be localized or anything
* Analytics events will come in a future PR

| Classic | Dark Contrast | Default Dark |
|---|---|---|
| ![Classic](https://github.com/user-attachments/assets/e92cb782-e536-4258-98dc-d4662972a599) | ![Dark Contrast](https://github.com/user-attachments/assets/113cebe2-95b6-40bb-a566-74164881fae9) | ![Default Dark](https://github.com/user-attachments/assets/b8a9e4c5-9e00-4b95-ab81-1f0e78609d7f) |


| Default Light | Electricity | Extra Dark |
|---|---|---|
| ![Default Light](https://github.com/user-attachments/assets/fa28b72a-13c9-4c3e-ac83-7e2b80286827) | ![Electricity](https://github.com/user-attachments/assets/3b94ae4f-c681-47d4-bb28-d8230026ee75) | ![Extra Dark](https://github.com/user-attachments/assets/ba4a5a0c-a2db-4805-990b-dd5c2029fdcd) |


| Indigo | Light Contrast | Radioactive | Rose
|---|---|---|---|
| ![Indigo](https://github.com/user-attachments/assets/df0a8668-ccd1-43a8-86d2-73bd399828a4) | ![Light Contrast](https://github.com/user-attachments/assets/45211cad-46fa-4dde-a3ad-c91a2212d374) | ![Radioactive](https://github.com/user-attachments/assets/4bdcd55e-9362-43d6-b03d-11e0135a3c3c) | ![Rosé](https://github.com/user-attachments/assets/45436a7e-c806-4f4b-907c-a81349ee2890) |

## To test

* Enable the `bannerAds` feature flag
* Restart the app
* Navigate to the Podcasts screen
* ✅ Ensure that the banner ad shows at the top

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

bjtitus committed:
Add BannerAdView

bjtitus committed:
Add Banner to podcast list

bjtitus committed:
Set banner ads feature flag to `false`

Until we have the network fetching of the ad, this should probably be false.

danielebogo reviewed (approved):
@bjtitus  working correctly. 

One thing I noticed though is that the banner doesn't expand if the text is longer than 3 lines. This also includes some dynamic font changes. Can we dynamically change the height and keep a min height?

bjtitus commented:
@danielebogo This is something I'll follow up on a bit more in a future PR but I've added a fixed size modifier so the text isn't truncated and top alignment. I need to do a few other minor tweaks to match Dom's accessibility designs and fix max font size and padding so I added a [tracking issue](https://github.com/Automattic/pocket-casts-ios/issues/3208).

Does this look like what you were thinking?

| Small | Large | Lots of Text |
|--|--|--|
| ![Simulator Screenshot - iPhone 16 Pro - 2025-06-04 at 08 34 37](https://github.com/user-attachments/assets/687a0ef5-3bff-42e4-a846-2a28c88a92bf) | ![Simulator Screenshot - iPhone 16 Pro - 2025-06-04 at 08 36 38](https://github.com/user-attachments/assets/34408225-c55d-4c66-8af4-013bf6b77983) | ![Simulator Screenshot - iPhone 16 Pro - 2025-06-04 at 08 34 06](https://github.com/user-attachments/assets/19ff08ca-5bd2-455d-bcae-b2621e11f085) |

bjtitus committed:
Add fixed vertical size to ad text

bjtitus committed:
Top align creative

danielebogo commented:
@bjtitus yep that's correct! Also a reminder in SwiftUI we can use `.font(size: 14.0, style: .body, weight: .medium)` modifier
